painttiming.bs

```diff
@@ -206,13 +206,17 @@ Reporting paint timing {#sec-reporting-paint-timing}
 <h4 dfn>Report paint timing</h4>
 
 <div algorithm="Report paint timing">
+    Each [=document=] has an associated array of previously reported paint types. Let that array be |reportedPaints|.
```

Comment:
It would be better if this was a [set](https://infra.spec.whatwg.org/#sets), rather than an array

Reply:
It would also be better if this set was not monkey-patched onto document here, but added to be part of HTML

Reply:
> It would be better if this was a [set](https://infra.spec.whatwg.org/#sets), rather than an array

OK.

Reply:
> It would also be better if this set was not monkey-patched onto document here, but added to be part of HTML

It seems like specs do this all the time; The same way https://w3c.github.io/performance-timeline/#dfn-frozen-array-of-supported-entry-types is not part of the global object spec.

Reply:
See for example https://wicg.github.io/ScrollToTextFragment/#parsing-the-fragment-directive
Fragment directive is associated with a document, but not mentioned in the HTML spec...

Reply:
The first example is a fair point (the second is a slightly different case). I'll know the HTML editors are not fans of algorithmic monkey-patching (and we should therefore avoid it). I'm now less sure the same applies to extending HTML objects (which is what this is doing). I'll ask.

Reply:
We're not really extending an HTML object - none of this is web-exposed. The document here is just something like a lifetime context for the set. But let's see what they say :)

Reply:
btw it might be better to associate the set with an eligible browsing context rather than with a document... let's see what we agree on  https://github.com/w3c/paint-timing/pull/79

Reply:
See new version @yoavweiss